core/src/com/ericc/the/game/systems/logic/ActionHandlingSystem.java

```diff
@@ -0,0 +1,174 @@
+                Mappers.currentAction.get(currentEntity).action = Mappers.intention.get(currentEntity).currentIntent;
+                updatePosition(currentEntity);
+            }
+
+            Mappers.intention.get(currentEntity).currentIntent = Actions.NOTHING;
+        }
+
+        movableInitiatives.clear();
+    }
+
+    private boolean analyzeMove(Entity entity) {
+        // This section will require expansion after more actions are added in.
+        IntentionComponent intent = Mappers.intention.get(entity);
+        if (intent.currentIntent instanceof MovementAction) {
+            return analyzeMovementAction(entity, intent);
+        }
+        return true;
+    }
+
+    private boolean analyzeMovementAction(Entity entity, IntentionComponent intent) {
+        PositionComponent pos = Mappers.position.get(entity);
+        MovementAction move = (MovementAction) intent.currentIntent;
+        int targetX, targetY;
+
+        targetX = ((move.direction == Direction.RIGHT) ? 1 : 0)
+                + ((move.direction == Direction.LEFT) ? -1 : 0)
+                + pos.x;
+        targetY = ((move.direction == Direction.UP) ? 1 : 0)
+                + ((move.direction == Direction.DOWN) ? -1 : 0)
+                + pos.y;
+
+        // If the target field is not passable movement is not possible
+        if (!map.isPassable(targetX, targetY)) {
+            if (entity instanceof Player) System.out.print("Player tried running into the wall.\n");
+            return false;
+        }
+
+        GridPoint2 targetPos = new GridPoint2(targetX, targetY);
+
+        Entity collidingEntity = interactives.get(targetPos);
+
+        if (collidingEntity != null) {
+            if (collidingEntity instanceof PushableObject) {
+                boolean canBePushed = analyzeMovementAction(collidingEntity, intent);
+
+                if (canBePushed) {
+                    Mappers.currentAction.get(collidingEntity).action = new MovementAction(move.direction);
+                    updatePosition(collidingEntity);
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+            if (collidingEntity instanceof Player) {
+                // Damage system will apply damage to the Player here
+                // if the entity is an agressive mob
+                return false;
+            }
+            if (collidingEntity instanceof Mob) {
+                // Damage system will apply damage to the Mob here if the entity
+                // is a player, or will not move if it is a mob or some other object
+                if (entity instanceof Player) System.out.print("Player tried running into a mob.\n");
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private void updatePosition(Entity entity) {
+        PositionComponent pos = Mappers.position.get(entity);
+        CurrentActionComponent action = Mappers.currentAction.get(entity);
```

Comment:
Czemu to się nie może dziać samo w następnej klatce na podstawie tego co zrobił movement system?

Reply:
Ponieważ zmiany dokonane w turze pojedynczego moba/playera muszą być widoczne natychmiast. Inaczej możemy wchodzić np. na to samo miejsce z dwóch sąsiednich bo oba entities patrzą na stan mapy sprzed ruchów wykonanych w tej turze.